### PR TITLE
Fix h2c submodule

### PR DESCRIPTION
**Pull request by armfazh**
Changes:
 - Removes duplicated submodule for h2c. Better to use the one inside the voprf module.
 - ~Exports more values to make the Fake case reproducible. No changes on test vectors, just more verbose info for the `Fake=True` cases.~

### CODE REVIEW

**Review by bytemare — APPROVED** [review]
Works for me ! Thank you :)

**Review by kevinlewi — CHANGES_REQUESTED** [review]
@armfazh: Thanks! I'm struggling to understand the need for the changes to make the fake case reproducible, though...

Why do we need to include the input parameters to generate KE1 (e.g. password, client_nonce, etc.), and view KE1 as an output (as opposed to an input)? Shouldn't KE1 be an input, and KE2 be the only output?

**Comment by armfazh** [comment]
It wasn't required at all. Test vectors are ok as is.
I just update this PR with changes only for h2c submodule.

**Review by kevinlewi — APPROVED** [review]
LGTM! Thanks!

File: poc/opaque_ake.sage [hunk 1]
```diff
@@ -61,7 +61,7 @@ class OPAQUE3DH(KeyExchange):
         return {
             "Name": "3DH",
             "Group": self.config.group.name,
-            "OPRF": to_hex(I2OSP(self.config.oprf_suite.identifier, 2)),
+            "OPRF": self.config.oprf_suite.identifier,
```
Comment (bytemare) [inline poc/opaque_ake.sage:64]: Good catch! I must have forgotten to commit that

**Review by chris-wood — APPROVED** [review]

**Comment by chris-wood** [comment]
Thanks, @armfazh! Merging.

**Pull request merged**